editor/editor_node.cpp

```diff
@@ -4437,6 +4437,7 @@ void EditorNode::_dock_floating_close_request(Control *p_control) {
 	dock_slot[window_slot]->add_child(p_control);
 	dock_slot[window_slot]->move_child(p_control, MIN((int)window->get_meta("dock_index"), dock_slot[window_slot]->get_tab_count()));
 	dock_slot[window_slot]->set_current_tab(window->get_meta("dock_index"));
+	dock_slot[window_slot]->set_tab_title(dock_slot[window_slot]->get_tab_idx_from_control(p_control), TTRGET(p_control->get_name()));
```

Comment:
This looks good, but the problem now is that we have two different ways to set the tab title which may diverge:
```
editor/editor_node.cpp
7008:   dock_slot[DOCK_SLOT_RIGHT_UL]->set_tab_title(dock_slot[DOCK_SLOT_RIGHT_UL]->get_tab_idx_from_control(InspectorDock::get_singleton()), TTR("Inspector"));
```

If someone updates this `TTR("Inspector")`, without updating the `set_name("Inspector")`, or the other way around, there will be a discrepancy.

Reply:
Now we can only make sure that dock title should match its node name. The dock system may require some refactoring in future...